### PR TITLE
Added __FlashStringHelper support for PROGMEM

### PR DESCRIPTION
**Pull request by brunnels**
Added in methods and macros to assist with PROGMEM usage

### CODE REVIEW

**Comment by anakod** [comment]
ICACHE_FLASH_ATTR isn't needed, because Sming building system locate all code to IROM by default. Instead you should mark with iram attrinbute to methods what should go to IRAM memory.

__FlashStringHelper will be very fine freature, I think.

**Comment by brunnels** [comment]
Okay, thanks for the info. I didn't know that. I'll modify this evening.

**Comment by brunnels** [comment]
Okay, should be good to go now.

File: Sming/Services/ArduinoJson/src/JsonBuffer.cpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 using namespace ArduinoJson;
 using namespace ArduinoJson::Internals;
 
-JsonStringStorage JsonStringStorage::_invalid(NULL);
+JsonStringStorage JsonStringStorage::_invalid("");
```
Comment (anakod) [inline Sming/Services/ArduinoJson/src/JsonBuffer.cpp]: Why not NULL or nullptr for invalid values?

Comment (brunnels) [inline Sming/Services/ArduinoJson/src/JsonBuffer.cpp]: NULL is macro'd to 0 which is int so throws a build error. You can try
nullptr and see if it builds but I think I tried it.

**Comment by kireevco** [comment]
@brunnels thank you so much for your PR. 
We are trying to implement git-flow (#230). Can you please re-submit your PR to a feature branch?
For example _feature/progmem-flashstring-helper_?
Also, please squash your commits to one - too many of them.

**Comment by hreintke** [comment]
@brunnels : 
We are restructuring and cleaning Sming repository and PR's.
Can you please squash your commits as some are reverting others and complete PR is unclear.
I have seen updates to ArduinoJson source. Can you explicitly mention why you want that update as it would diverge form the used library. 
PS There are plans to upgrade to a newer version of ArduinoJson.

**Comment by hreintke** [comment]
This issue has been inactive for a long period and is being closed. If the issue is still valid please open a new with complete description

**Pull request closed without merging**